### ISSUE TITLE
`cabal install policeman` fails using cabal version 3.6

### ISSUE THREAD

**Issue by jneira** [kowainik/policeman#65]
* Hi, i've just tried to install policeman with `cabal  --store-dir=path/to/fresh/store install policeman -w ghc-8.8.4` and it failed using my recently installed cabal-install-3.6.0.0 with:

```
Installing library in D:\cabal\store\ghc-8.8.4\incoming\new-15164\cabal\store\ghc-8.8.4\relude-0.6.0.0-68df4281a545941c3074b5adeb0c0bf13b9935a2\lib
Configuring library for policeman-0.0.0.0..
Preprocessing library for policeman-0.0.0.0..
Building library for policeman-0.0.0.0..
[ 1 of 15] Compiling Paths_policeman

dist\build\autogen\Paths_policeman.hs:66:22: error:
    * 'last' works with 'NonEmpty', not ordinary lists.
      Possible fix:
          Replace: [Char]
          With:    NonEmpty Char

      However, you can use 'last' with the ordinary lists.
      Apply 'viaNonEmpty' function from relude:
          viaNonEmpty last (yourList)
      Note, that this will return 'Maybe Char'
      therefore it is a safe function unlike 'last' from the standard Prelude
    * In the first argument of `isPathSeparator', namely `(last dir)'
      In the expression: isPathSeparator (last dir)
      In a stmt of a pattern guard for
                     an equation for `joinFileName':
        isPathSeparator (last dir)
   |
66 |   | isPathSeparator (last dir) = dir ++ fname
   |                      ^^^^^^^^
cabal-3.6.0.0.exe: Failed to build policeman-0.0.0.0 (which is required by
exe:policeman from policeman-0.0.0.0).
```

* `cabal-3.4.0.0 install policeman -w ghc-8.8.4` works and after that, cabal-3.6.0.0 also works so you need a fresh store dir to reproduce

**Comment by Mikolaj** [comment]
It seems the breakage stems from the use of `last` in Paths_ and  the fact that Prelude is replaced in policeman. We (the cabal project) would welcome a PR that changes Paths_ to use the standard Prelude regardless which method is used to overwrite it (see https://hackage.haskell.org/package/relude for a few methods). For now, we don't see an easy workaround, except a rather crude one, namely using `NoImplicitPrelude` in each non-autogenerated file as the way to replace Prelude. Let us know how we can help.

Edit: PR with tests so that the breakage is caught before cabal release.

**Comment by Mikolaj** [comment]
See also the approach from https://github.com/haskell/cabal/pull/7646 (not using Prelude at all in Paths_), which seems easier to do and test.

**Comment by amesgen** [comment]
I also stumbled upon this issue with cabal 3.6, and FTR, another workaround when using the `Paths_` module for the version number is to use the CPP variable `CURRENT_PACKAGE_VERSION` which has been available since cabal 2.0: https://cabal.readthedocs.io/en/3.6/cabal-package.html?highlight=CURRENT_PACKAGE_VERSION#conditional-compilation

**Comment by chshersh** [comment]
`policeman` now builds with `cabal-install-3.6.2.0` after the corresponding cabal issue was resolved. So I'm closing this one.

Thanks, @alexbiehl and the Cabal team for fixing the problem!
